src/kernel/arch/x86/arch.zig

```diff
@@ -45,3 +57,85 @@ pub fn inb(port: u16) u8 {
 pub fn ioWait() void {
     outb(0x80, 0);
 }
+
+// Can't use gdt.KERNEL_DATA_OFFSET in the assembly bellow, so have to store in a constant.
+const KERNEL_DATA_OFFSET: u16 = gdt.KERNEL_DATA_OFFSET;
+const TSS_OFFSET: u16 = gdt.TSS_OFFSET;
+
+///
+/// Load the GDT and refreshing the code segment with the code segment offset of the kernel as we
+/// are still in kernel land. Also loads the kernel data segment into all the other segment
+/// registers.
+///
+/// Arguments:
+///     IN gdt_ptr: *gdt.GdtPtr - The address to the GDT.
+///
+pub fn lgdt(gdt_ptr: *const gdt.GdtPtr) void {
+    // Load the GDT into the CPU
+    asm volatile ("lgdt (%%eax)" : : [gdt_ptr] "{eax}" (gdt_ptr));
+    // Load the kernel data segment, index into the GDT
+    asm volatile ("mov %%bx, %%ds" : : [KERNEL_DATA_OFFSET] "{bx}" (KERNEL_DATA_OFFSET));
+    asm volatile ("mov %%bx, %%es");
+    asm volatile ("mov %%bx, %%fs");
+    asm volatile ("mov %%bx, %%gs");
+    asm volatile ("mov %%bx, %%ss");
+    // Load the kernel code segment into the CS register
+    asm volatile (
+        \\ljmp $0x08, $1f
+        \\1:
+    );
+}
+
+///
+/// Load the TSS into the CPU.
+///
+pub fn ltr() void {
+    asm volatile ("ltr %%ax" : : [TSS_OFFSET] "{ax}" (TSS_OFFSET));
+}
+
+/// Load the IDT into the CPU.
+pub fn lidt(idt_ptr: *const idt.IdtPtr) void {
+    asm volatile ("lidt (%%eax)" : : [idt_ptr] "{eax}" (idt_ptr));
+}
+
+///
+/// Enable interrupts
+///
+pub fn sti() void {
```

Comment:
`halt`, `sti` and `cli` are x86-specific names so more architecture-agnostic names would be `halt`, `enableInterrupts` and `disableInterrupts`.

Reply:
Can do, just was making it clear that these are x86 instructions instead of functions